cumulusci/tasks/bulkdata/utils.py

```diff
@@ -85,6 +84,12 @@ def _parse_job_state(self, xml):
         elif "Failed" in statuses:
             return "Failed", state_messages
 
+        failures = tree.find(f".//{{{ns}}}numberRecordsFailed")
+        if failures is not None:
+            num_failures = int(failures.text)
+            if num_failures:
+                return "Failed", [f"Failures detected: {num_failures}"]
```

Comment:
This appears to break existing functionality in `LoadData` that allows row-level errors to be ignored at user option via setting `ignore_row_errors` to `True`. I haven't done an E2E test though.

Reply:
I think I've addressed this now. `LoadData` still doesn't use the error messages returned from the lower level, but you would know better than I do whether the same errors will be discovered through another channel.

Reply:
@prescod you've managed to make it so that LoadData will actually process the results, and raise or not for record-level errors based on the ignore_row_errors setting. But you've changed what state _parse_job_state returns in the case where all batches completed but some records failed. Since it now returns "Failed" in that case instead of "Completed", this line will now raise an exception and prevent processing of other mappings: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/tasks/bulkdata/load.py#L86

The point of the ignore_row_errors=True option is to allow further mappings to be processed in the case where a small number of rows failed but most succeeded. I think I added this because I was doing a load where one object was prone to row lock errors (it was the load of GAU allocations where one of the GAUs had thousands of related allocations) and it was preferable to complete the data load with some records missing than to abort, because the whole thing took many hours to run.

I'd be willing to add a test for the correct behavior with regard to ignore_row_errors, but may not be able to get to it for a few days.

Reply:
Half of the goal of the PR was to provoke conversation, because this behaviour is literally not backwards compatible in any context. For all I know, it will cause many of my Performance Tests which previously claimed to succeed to now claim to fail. It may be a messy process to determine which of the failures are acceptable and which are not.

I guess we should have states for Completed, Failed and CompletedWithErrors. Then the caller can decide how serious a CompletedWithErrors situation is.